### DISCUSSION TITLE
Nothing Happens after Flashing

### DISCUSSION THREAD

**Discussion by Tinteex** [jomjol/AI-on-the-edge-device#1182] [Q&A]
### The Problem

Im using 4 ESP for my meters already, they all work flawlessly.

Today i wanted to setup a fifth one, but after flashing the firmware via the esp tool (no errors, it all works flawlessly) and entering the sd card with the correct credentials, nothing happens. no led is flashing, expcept the basic power one.

things i tried

- change sd card (tried 10 different ones, from 256mb to 32gb)
- try different wifi network (tried 3)
- changed cables & psu
- clarified that it has 4mb psram mininum
- tried multiple different camera modules

nothing seems to work. the log on the sd card is empty.

i tried flashing it via homeassistant with another firmware to check if the wifi module is working, and i can confirm that it is.

i also tried 3 other esp-32 modules.... none of them will work. (however they did when i first set them up a few months ago...)

does anyone have an idea what could cause this issue?

it makes me lose my mind, since i got 4 others that work fine.

### Version

12.0.1

**Comment by jomjol** [comment]
Did you just put in one of the working sd-cards into the non workin?

Next step would be to make a log file of the serial output during startup and post this (not the sd-card log, but the output of the RX/TX)

**Comment by Tinteex** [comment]
Hello.

I tried that aswell as using fresh sd cards of different sizes to try.

when i put the device to my ear it sounds like it is restarting After 2 seconds.


do you have any printers on how i could read out the rt/tx log? do i need a special device? 

thank you.

**Comment by caco3** [comment]
Use Putty to get the UART log.
When you power your device through the USB connector (like for the initial flashing), you can set Putty to the same port and let it listen to it.

**Comment by caco3** [comment]
See https://docs.espressif.com/projects/esp-idf/en/latest/esp32/get-started/establish-serial-connection.html resp. https://www.eye4software.com/hydromagic/documentation/articles-and-howtos/serial-port-logging/

**Comment by Tinteex** [comment]
Thank you! i will check it out asap

**Comment by Tinteex** [comment]
This is what i get using Putty
https://prnt.sc/sE-cP6VLxct7

**Reply by caco3** [reply]
See https://github.com/jomjol/AI-on-the-edge-device/wiki/Installation#wiring

If GPIO0 gets pulled down, it starts in Bootloader Mode

**Comment by caco3** [comment]
So you want to say there is no output at all?
That means either you did not connect it correctly, or the firmware does not start at all.

> i tried flashing it via homeassistant with another firmware to check if the wifi module is working, and i can confirm that it is.

How can homeassistant be used to flash firmware? Please use the officially documented tools, see https://github.com/jomjol/AI-on-the-edge-device/wiki/Installation!

> i also tried 3 other esp-32 modules.... none of them will work. (however they did when i first set them up a few months ago...)

I don't think then its a problem of the  hardware or software!

**Comment by Tinteex** [comment]
you can use esp home to flash esp boards, i flashed it with different firmware to make sure the board isnt fried. 

like i said it was a basic firmware, not ai-on-the-edge.

i made sure it is port COM8 and used putty to connect, sadly no output

i got an output now.

diagonal lines lol

https://prnt.sc/sEoDxnKz1rig

**Comment by caco3** [comment]
are you sure it is the right COM?
And you are not in bootloader mode?

**Comment by Tinteex** [comment]
yes, i double checked the right com port.

how can i check that im not running bootloader mode?

https://prnt.sc/bdFbsciJDXA2

**Comment by jomjol** [comment]
I sometimes also have problems with Putty (don't know why).
An easy to use serial monitor is also in the arduino programming interface. It works also nicely for the esp32 (assuming the right interface setting).

**Comment by Tinteex** [comment]
> I sometimes also have problems with Putty (don't know why). An easy to use serial monitor is also in the arduino programming interface. It works also nicely for the esp32 (assuming the right interface setting).

my output from arduino

https://prnt.sc/QeH04pbpvZc3

**Comment by jomjol** [comment]
What is the setting of your serial (baud rate, ...)?

**Comment by Tinteex** [comment]
> 

i tried all baudrate settings, they return nothing, however 9600 gave me the garbled questionmarks.

currently i dont get anything anymore.
